codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/testutil/TestHelpers.kt

```diff
@@ -94,6 +94,12 @@ fun String.asSmithyModel(sourceLocation: String? = null): Model {
         .unwrap()
 }
 
+fun String.asSmithyModel_Version2(sourceLocation: String? = null): Model {
```

Comment:
Kotlin has defaulted arguments, so you could reuse the above `asSmithyModel`:
```kotlin
fun String.asSmithyModel(sourceLocation: String? = null, smithyVersion: String = SmithyVersion): Model { ... }
```

Then when invoking it:
```kotlin
"some model".asSmithyModel() // 1.0
"some model".asSmithyModel(smithyVersion = "2.0") // 2.0
```

Could potentially make an enum to avoid hardcoding string version numbers everywhere.